crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -193,17 +245,46 @@ mod orderbook {
         fn _create_listing_order(
             ref self: ContractState, order: OrderV1, order_type: OrderType, order_hash: felt252
         ) {
+            let token_hash = order.compute_token_hash();
+            let current_listing_orderhash = self.token_listings.read(token_hash);
+            let current_order_status = order_status_read(current_listing_orderhash);
+            // if there is an order already, we need to cancel it
+            if current_listing_orderhash.is_non_zero() {
+                // if current order has a status, we need to cancel it
+                if current_order_status.is_some() {
+                    let current_order_status = current_order_status.unwrap();
+                    // if order is already fulfilled, we cannot cancel it
+                    if (current_order_status == OrderStatus::Fulfilled) {
+                        panic_with_felt252(orderbook_errors::ORDER_FULFILLED);
+                    } else {
+                        // change old order status to cancelled
+                        order_status_write(
+                            current_listing_orderhash, OrderStatus::CancelledByNewOrder
+                        );
+                    }
+                } else {
+                    // if order status is not found, panic
+                    panic_with_felt252(orderbook_errors::STATUS_NOT_FOUND);
+                }
+            }
+            // Associate order_hash to token_hash on the storage
+            order_write(order_hash, order_type, order);
+            // Change order status to fulfilled
+            order_status_write(current_listing_orderhash, OrderStatus::Fulfilled);
+            // Write new order with status open
+            self.token_listings.write(token_hash, order_hash);
```

Comment:
To be more explicit, perhaps this function of the database may take the status as input. This will make the call very explicit. WDYT?